### PR TITLE
feat: cookie banner and Askem feedback buttons (HL-909, HL-922)

### PR DESCRIPTION
**Pull request by mjturt**
## Description :sparkles:

- Add cookie consent bar
- Add Askem feedback buttons

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe applicant result is __failed__ for https://helsinkilisa-ui-pr2550.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  };
+
+  const askemHostname = 'reactandshare.com';
+
+  type CookieTexts = {
+    title: { fi: string; sv: string; en: string };
+    askemDescription: { fi: string; sv: string; en: string };
+  };
+
+  const text: CookieTexts = {
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx:31]: Shouldn't these be in locale JSON files?

File: frontend/benefit/applicant/src/components/applications/Applications.sc.ts [hunk 2]
```diff
@@ -49,3 +49,24 @@ export const $SpinnerContainer = styled.div`
...
+export const $AskemContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  min-height: 129px;
+  margin: 0;
+  padding-left: ${(props) => props.theme.spacing.l};
+  padding-right: ${(props) => props.theme.spacing.l};
+  ${respondAbove('sm')`
+    flex-direction: row;
+
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/components/applications/Applications.sc.ts:62]: Added empty row before `${respondAbove('sm')` and empty row removed after `flex-direction: row;` would be more readable.

File: frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    void axios.get(BackendEndpoint.USER_OPTIONS, {
+      params: { lang: newLanguage },
+    });
+
+    void router.push({ pathname, query }, asPath, {
+      locale: newLanguage,
+    });
+  };
+
+  const askemHostname = 'reactandshare.com';
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx:24]: Should this be under `constants.ts`?

File: frontend/benefit/applicant/src/components/notificationView/NotificatinsView.sc.ts [hunk 4]
```diff
@@ -5,14 +5,13 @@ import styled from 'styled-components';
...
-  margin: ${(props) => props.theme.spacing.xl} 0
-    ${(props) => props.theme.spacing.xl} 0;
+  margin-top: ${(props) => props.theme.spacing.xl};
   padding: ${(props) => props.theme.spacing.l};
   background: ${(props) => props.theme.colors.fogLight};
 
   ${respondAbove('sm')`
     flex-direction: row;
-  
+
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/components/notificationView/NotificatinsView.sc.ts:14]: Empty row here as well.

File: frontend/benefit/applicant/src/hooks/useAnalytics.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+): boolean => {
+  const showAskem = canShowAskem(lang);
+  useEffect(() => {
+    if (!canShowAskem || isLoading) {
+      return () => {};
+    }
+
+    const script = document.createElement('script');
+    // eslint-disable-next-line scanjs-rules/assign_to_src
+    script.src = 'https://cdn.reactandshare.com/plugin/rns.js';
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/hooks/useAnalytics.ts:18]: Should be under contants?

**Review by sirtawast — CHANGES_REQUESTED** [review]

File: frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx (hunk 1 above)
Comment (mjturt) [reply frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx:31]: Definitely yes, but just could not make it work. The HDS cookie banner has some own localizations that mess up with ours. Saw same kind of a solution here <https://github.com/City-of-Helsinki/employment-services-ui/blob/08cbb83303ca96094d60b4cbb0d4678010b0fc12/src/hooks/useAnalytics.ts#L33>

File: frontend/benefit/applicant/src/components/applications/Applications.sc.ts (hunk 2 above)
Comment (mjturt) [reply frontend/benefit/applicant/src/components/applications/Applications.sc.ts:62]: I will try to improve it. Just copied this from the `NotificationView` component, as I needed the same layout here.

File: frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx (hunk 3 above)
Comment (mjturt) [reply frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx:24]: Yes, definitely

File: frontend/benefit/applicant/src/hooks/useAnalytics.ts (hunk 5 above)
Comment (mjturt) [reply frontend/benefit/applicant/src/hooks/useAnalytics.ts:18]: True

File: frontend/benefit/applicant/src/components/notificationView/NotificatinsView.sc.ts (hunk 4 above)
Comment (mjturt) [reply frontend/benefit/applicant/src/components/notificationView/NotificatinsView.sc.ts:14]: My editor just removed unnecessary indents here, no new newlines added. (and we should have a newline character as well after the last line (unix-style) as mentioned in our `.editorconfig.`)

**Comment by mjturt** [comment]
> Tests are not passing. The cookie bar should be hidden for tests, preferably using a flag.

Thank you, I will try!

File: frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx (hunk 1 above)
Comment (sirtawast) [reply frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx:31]: Hmm... suppouse you can try importing json as such and just stick the objects in like I've done in tests.

Comment (sirtawast) [reply frontend/benefit/applicant/src/components/cookieConsent/CookieConsent.tsx:31]: For example, 

https://github.com/City-of-Helsinki/yjdh/blob/main/frontend/benefit/applicant/browser-tests/page-model/deminimis.ts#L3

and

https://github.com/City-of-Helsinki/yjdh/blob/main/frontend/benefit/applicant/browser-tests/page-model/deminimis.ts#L137

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2550.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2550.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2550.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**